### ISSUE TITLE
Update from file

### ISSUE THREAD

**Issue by svnpenn** [FirefoxBar/xStyle#45]
My Update URL is this:

~~~
file:///D:/Git/bay-master/config/xstyle/private.css
~~~

but it does not work

**Comment by sylingd** [comment]
WebExtension can not access local file

**Comment by svnpenn** [comment]
:(

**Comment by svnpenn** [comment]
@sylingd 

http://stackoverflow.com/a/44516256

**Comment by sylingd** [comment]
I will test it

**Comment by sylingd** [comment]
I tested it, but both Chrome and Firefox not work
![2017-11-20_122416](https://user-images.githubusercontent.com/5326684/33002252-f65e80c6-cded-11e7-9311-b12402c49d43.png)
![2017-11-20_122456](https://user-images.githubusercontent.com/5326684/33002253-f84f8038-cded-11e7-8dce-d7c589bacbbe.png)

**Comment by sylingd** [comment]
Chrome can give extensions permission to read local files by check it in `chrome://extensions/`
![2017-11-20_122828](https://user-images.githubusercontent.com/5326684/33002331-778c702c-cdee-11e7-988d-843d0ece980f.png)
And it works well with `XMLHttpRequest`
![2017-11-20_122810](https://user-images.githubusercontent.com/5326684/33002342-8a5a4fee-cdee-11e7-99d8-40c939ad7e36.png)
But Firefox doesn't have the same switch

**Comment by svnpenn** [comment]
works for me with the Browser console:

~~~
function readFile(_path, _cb){

    fetch(_path, {mode:'same-origin'})   // <-- important

    .then(function(_res) {
        return _res.blob();
    })

    .then(function(_blob) {
        var reader = new FileReader();

        reader.addEventListener("loadend", function() {
            _cb(this.result);
        });

        reader.readAsText(_blob); 
    });
};
undefined

readFile('file:///D:/Desktop/test.txt', function(_res){

    console.log(_res); // <--  result (file content)

});
undefined
hello world
~~~

**Comment by sylingd** [comment]
Tested on nightly 59.0a1 2017-11-20, maybe Firefox have fixed this "BUG"
![2017-11-20_125115](https://user-images.githubusercontent.com/5326684/33002829-8c82546c-cdf1-11e7-9bba-3b95cdac689d.png)

**Comment by svnpenn** [comment]
no, you must do it with the Browser Console, not the Web Console

**Comment by sylingd** [comment]
But the runtime environment of WebExtensions is not same as Browser Console, it's same as Web Console

**Comment by svnpenn** [comment]
@sylingd yes it does work:

http://github.com/Tampermonkey/tampermonkey/issues/347#issuecomment-345539557

**Comment by sylingd** [comment]
I have asked a staff of Mozilla China, He said that this is a BUG, and they have already marked it as "P1"importance
(Priority: This field describes the importance and order in which a bug should be fixed compared to other bugs. This field is utilized by the programmers/engineers to prioritize their work to be done where P1 is considered the highest and P5 is the lowest. See Bugmasters/Process/Triage or for the main Buzilla project, Bugzilla:Priority_System.)

![qq 20171124213124](https://user-images.githubusercontent.com/5326684/33212614-f7d8c51e-d15e-11e7-9da4-83afcac0e516.png)

**Comment by svnpenn** [comment]
not sure where you got that picture - but not that is not a valid bug:

http://bugzilla.mozilla.org/buglist.cgi?quicksearch=webextensions+local

**Comment by sylingd** [comment]
I am one of the administrators of FirefoxBar(a big chinese firefox forum). We are stay in touch with Mozilla China.
This picture is sent by one of the staffs of Mozilla China. You can not view it because it is a private BUG.

**Comment by sylingd** [comment]
I found a extension called [GhostText](https://github.com/GhostText/GhostText), maybe it is helpful on writing styles. I am trying to add it into xStyle

**Comment by okyr-ux** [comment]
Hey, @sylingd, thanks for a helpful extension!
I have the same issue: styles don't autoupdate from local CSS files.
Is there any progress on that? Does it work in Firefox (Quantum)?

**Comment by sylingd** [comment]
@okyr-ux Until now, there is no legal API to read local files
